brave/src/main/java/brave/Tracer.java

```diff
@@ -495,6 +492,46 @@ public ScopedSpan startScopedSpan(String name) {
     return startScopedSpanWithParent(name, null);
   }
 
+  /**
+   * Like {@link #startScopedSpan(String)} except when there is no trace in process, the sampler
+   * {@link SamplerFunction#trySample(Object) triggers} against the supplied argument.
+   *
+   * @param name the {@link Span#name(String) span name}
+   * @param samplerFunction invoked if there's no {@link CurrentTraceContext#get() current trace}
+   * @param arg parameter to {@link SamplerFunction#trySample(Object)}
+   * @see #nextSpan(SamplerFunction, Object)
+   * @since 5.8
+   */
+  public <T> ScopedSpan startScopedSpan(String name, SamplerFunction<T> samplerFunction, T arg) {
+    if (name == null) throw new NullPointerException("name == null");
+    return newScopedSpan(name, nextContext(samplerFunction, arg));
+  }
+
+  /**
+   * Like {@link #nextSpan()} except when there is no trace in process, the sampler {@link
+   * SamplerFunction#trySample(Object) triggers} against the supplied argument.
+   *
+   * @param samplerFunction invoked if there's no {@link CurrentTraceContext#get() current trace}
+   * @param arg parameter to {@link SamplerFunction#trySample(Object)}
+   * @see #startScopedSpan(String, SamplerFunction, Object)
+   * @see #nextSpan(TraceContextOrSamplingFlags)
+   * @since 5.8
+   */
+  public <T> Span nextSpan(SamplerFunction<T> samplerFunction, T arg) {
+    return _toSpan(nextContext(samplerFunction, arg));
+  }
+
+  <T> TraceContext nextContext(SamplerFunction<T> samplerFunction, T arg) {
+    if (samplerFunction == null) throw new NullPointerException("samplerFunction == null");
+    if (arg == null) throw new NullPointerException("arg == null");
```

Comment:
If `SamplerFunction` is intended for use by users, it might make sense to let them accept `null` if they want it. Some pattern like a function of an attribute, with arg being an extracted attribute that could be `null`, seems like a reasonable pattern

Reply:
that's ok with me.

Reply:
added code PTAL

Reply:
I think the new tests verify the default matchers accept null but don't we still need to remove this check so users can call this method with null? How would instrumentation pass null to the matcher?

Reply:
sorry I forgot to push most of the commit :P